M2/Macaulay2/m2/programs.m2

```diff
@@ -95,11 +95,13 @@ findProgram(String, List) := opts -> (name, cmds) -> (
     pathsToTry = append(pathsToTry, prefixDirectory | currentLayout#"programs");
     -- any additional paths specified by the caller
     pathsToTry = pathsToTry | opts.AdditionalPaths;
-    -- finally, try PATH
+    -- try PATH
     if getenv "PATH" != "" then
 	pathsToTry = join(pathsToTry,
 	    apply(separate(":", getenv "PATH"), dir ->
 		if dir == "" then "." else dir));
+    -- try directory containing M2-binary
+    pathsToTry = append(pathsToTry, replace("/M2-binary", "", commandLine#0));
```

Comment:
If you start M2 with `M2-binary`, then the `commandLine` is just `M2-binary`, which might not work here. Replacing `"/*M2-binary"` might be better?

Reply:
Yeah, I wondered about this.  But if `commandLine#0` is just `M2-binary`, then how can we recover the directory?  Just using `replace` would leave us with the empty string...

This is also a bug in `run.m2`:

https://github.com/Macaulay2/M2/blob/e202a19c605bc14338147549942ae73286291939/M2/Macaulay2/m2/run.m2#L100

```m2
profzoom@peg-amy:~$ M2-binary
Macaulay2, version 1.18.0.1
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems,
               LLLBases, MinimalPrimes, PrimaryDecomposition, ReesAlgebra,
               Saturation, TangentCone

i1 : importFrom_Core {"runString"};

i2 : runString("2 + 2", Core, true)

 ulimit -c unlimited; ulimit -t 700; ulimit -m 850000; ulimit -s 8192; ulimit -n 512;  cd /tmp/M2-3501062-0/1-rundir/; GC_MAXIMUM_HEAP_SIZE=400M "/home/profzoom/M2-binary" --int --no-randomize --no-readline --silent --stop --print-width 77 <"/tmp/M2-3501062-0/0.m2" >>"/tmp/M2-3501062-0/0.tmp" 2>&1
/tmp/M2-3501062-0/0.tmp:0:1: (output file) error: Macaulay2 exited with status code 127
/tmp/M2-3501062-0/0.m2:0:1: (input file)
M2: *** Error 127

o2 = false

i3 : get "/tmp/M2-3501062-0/0.tmp"

o3 = -- -*- M2-comint -*- hash: 47171727
     sh: 1: /home/profzoom/M2-binary: not found
```

Reply:
We won't get the directory, but it should give an error either.

We get the executable somewhere in `startup.m2`, perhaps it should be kept.

Reply:
I've just pushed a new version where we use `bindir` from `startup.m2` to get the directory.